doc/how_to/custom_components/esm/custom_layout.md

```diff
@@ -234,149 +229,210 @@ A Panel `Column` or `Row` works as a list of objects. It is *list-like*. In this
 
-        layoutable_params = {name: self.param[name] for name in Layoutable.param}
-        self._layout = pn.Column(
-            **layoutable_params,
-        )
-        self._objects()
-
-    def __panel__(self):
-        return self._layout
-
-    @pn.depends("objects", watch=True)
-    def _objects(self):
-        objects = []
-        for object in self.objects:
-            objects.append(object)
-            objects.append(
-                pn.pane.HTML(
-                    styles={"width": "calc(100% - 15px)", "border-top": "3px dotted #bbb"},
-                    height=10,
-                )
-            )
-
-        self._layout[:] = objects
-
-
-ListLikeLayout(
-    "I love beat boxing",
-    "https://upload.wikimedia.org/wikipedia/commons/d/d3/Beatboxset1_pepouni.ogg",
-    "Yes I do!",
+      const splitDiv = document.createElement('div');
+      splitDiv.className = 'split';
+      splitDiv.style.height = `calc(100% - ${(objects.length - 1) * 10}px)`;
+
+      let splits = [];
+
+      objects.forEach((object, index) => {
+        const split = document.createElement('div');
+        splits.push(split)
+
+        splitDiv.appendChild(split);
+        split.appendChild(object);
+      })
+
+      Split(splits, {direction: 'vertical'})
+
+      return splitDiv
+    }"""
+
+    _stylesheets = [CSS]
+
+
+pn.extension("codeeditor")
+
+grid_js = GridJS(
+    pn.widgets.CodeEditor(
+        value="I love beat boxing\n" * 10, theme="monokai", sizing_mode="stretch_both"
+    ),
+    pn.panel(
+        "https://upload.wikimedia.org/wikipedia/commons/d/d3/Beatboxset1_pepouni.ogg",
+        sizing_mode="stretch_width",
+        height=100,
+    ),
+    pn.widgets.CodeEditor(
+        value="Yes I do!\n" * 10, theme="monokai", sizing_mode="stretch_both"
+    ),
     styles={"border": "2px solid lightgray"},
+    height=800,
+    width=500,
+    sizing_mode="fixed",
 ).servable()
 ```
 
-You must list `NamedListLike, Layoutable, Viewer` in exactly that order when you define the class! Other combinations might not work.
+You must list `NamedListLike, JSComponent` in exactly that order when you define the class! The other
+way around `JSComponent, NamedListLike` will not work.
 
 :::
 
-:::{tab-item} `JSComponent`
+:::{tab-item} `ReactComponent`
 
 ```{pyodide}
 import panel as pn
 import param
-from panel.custom import JSComponent
+from panel.custom import ReactComponent
 from panel.layout.base import NamedListLike
 
-pn.extension()
+CSS = """
+.gutter {
+    background-color: #eee;
+    background-repeat: no-repeat;
+    background-position: 50%;
+}
+.gutter.gutter-vertical {
+    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAFAQMAAABo7865AAAABlBMVEVHcEzMzMzyAv2sAAAAAXRSTlMAQObYZgAAABBJREFUeF5jOAMEEAIEEFwAn3kMwcB6I2AAAAAASUVORK5CYII=');
+    cursor: row-resize;
+}
+"""
 
 
-class ListLikeLayout(NamedListLike, JSComponent):
+class GridReact(NamedListLike, ReactComponent):
```

Comment:
The `GridReact` component can render initially. But when I append and pop I get error.

![image](https://github.com/user-attachments/assets/916ddd87-7f8a-45c7-9989-b7b3ce125918)

```python
import panel as pn
import param
from panel.custom import ReactComponent
from panel.layout.base import NamedListLike

CSS = """
.gutter {
    background-color: #eee;
    background-repeat: no-repeat;
    background-position: 50%;
}
.gutter.gutter-vertical {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAFAQMAAABo7865AAAABlBMVEVHcEzMzMzyAv2sAAAAAXRSTlMAQObYZgAAABBJREFUeF5jOAMEEAIEEFwAn3kMwcB6I2AAAAAASUVORK5CYII=');
    cursor: row-resize;
}
"""


class GridReact(NamedListLike, ReactComponent):
    objects = param.List()

    _esm = """
    import Split from 'https://esm.sh/react-split@2.0.14'

    export function render({ model}) {
      const objects = model.get_child("objects")
      console.log(objects)
      const calculatedHeight = `calc( 100% - ${(objects.length - 1) * 10}px )`;
      
      return (
        <Split
            className="split"
            direction="vertical"
            style={{ height: model.height }}
        >{...objects}</Split>
    )
    }"""

    _stylesheets = [CSS]


pn.extension("codeeditor")

grid_js = GridReact(
    pn.widgets.CodeEditor(
        value="I love beat boxing\n" * 10, theme="monokai", sizing_mode="stretch_both"
    ),
    pn.panel(
        "https://upload.wikimedia.org/wikipedia/commons/d/d3/Beatboxset1_pepouni.ogg",
        sizing_mode="stretch_width",
        height=100,
    ),
    pn.widgets.CodeEditor(
        value="Yes I do!\n" * 10, theme="monokai", sizing_mode="stretch_both"
    ),
    styles={"border": "2px solid lightgray"},
    height=800,
    width=500,
    sizing_mode="fixed",
).servable()


def update():
    grid_js.append(
        pn.widgets.CodeEditor(
            value="Another one bites the dust\n" * 10,
            theme="monokai",
            sizing_mode="stretch_both",
        )
    )
    grid_js.pop(-1)


pn.state.add_periodic_callback(update, 1000, count=3)
```

Reply:
Just double checking but was this:

```python
    grid_js.append(
        pn.widgets.CodeEditor(
            value="Another one bites the dust\n" * 10,
            theme="monokai",
            sizing_mode="stretch_both",
        )
    )
    grid_js.pop(-1)
```

just done to trigger the error?

Reply:
In any case, fixed here: https://github.com/holoviz/panel/pull/7159

Reply:
> Just double checking but was this:
> 
> ```python
>     grid_js.append(
>         pn.widgets.CodeEditor(
>             value="Another one bites the dust\n" * 10,
>             theme="monokai",
>             sizing_mode="stretch_both",
>         )
>     )
>     grid_js.pop(-1)
> ```
> 
> just done to trigger the error?

Yes

Reply:
Fixed.